src/hamster/edit_activity.py

```diff
@@ -302,8 +298,8 @@ def update_fields(self):
         self.start_time.time = self.fact.start_time
         self.end_time.time = self.fact.end_time
         self.end_time.set_start_time(self.fact.start_time)
-        self.start_date.date = self.fact.start_time
-        self.end_date.date = self.fact.end_time
+        self.start_date.date = self.fact.date or self.date
+        self.end_date.date = self.fact.date or self.date
```

Comment:
@gsobczyk, Why is this change needed? It has a bug, since it uses `self.fact.date` also for the start date, but I also wonder why this is needed at all. Especially the `or self.data` part seems to set a default, but why not just let (especially) the end date unset when none is set yet?